### PR TITLE
ARF-86187 update steps 3 & 4 to v3 designs

### PR DESCRIPTION
**Pull request by tjheffner**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
**ARF step 3 updates:**
- Updated intro page text
- Added hint text to supervisor name
- Phone Extension field is shortened width
- Added additional page to array builder for `Reason for leaving [employer]`
- Adds additional page after array builder pages for employment activities
- Updates date format on the card review

For step 3, I did not deal with the checkbox logic. The checkboxes have their own tickets (and will keep this review more focused)

**ARF step 4 updates:**
- Updated intro text
- Changed degree field to radio button
- Updated description text at top of main page
- Added additional text fields for degree, reason for not completing studies. These fields should be dependent on the radio button field, in both required status and visibility.
- Updated review page title
- Updates date format on the card review

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/86187

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by williamphelps13 — COMMENTED** [review]
Thanks for reaching out about this PR! I went through the PR in pretty good detail! Hopefully the notes are helpful and the solution for the conditional logic you reached out about is looking good! Happy to pair as well! I can run through this PR again when you move it out of draft!

File: src/applications/accreditation/21a/pages/03-employment-information-chapter/employersPages.js
Comment (tjheffner) [reply src/applications/accreditation/21a/pages/03-employment-information-chapter/employersPages.js:26]: I called this out in my PR description, that we should be using the `formatReviewDate` you added in the [Steps 1 & 2 PR](https://github.com/department-of-veterans-affairs/vets-website/pull/31019/files#diff-34c7bb72d1fddd213368717383f4d84945513334580e2fa823e91dd7ec4e8926), once it is merged

File: src/applications/accreditation/21a/pages/03-employment-information-chapter/employersPages.js
Comment (tjheffner) [reply src/applications/accreditation/21a/pages/03-employment-information-chapter/employersPages.js:189]: Sure, I can break it out

File: src/applications/accreditation/21a/pages/03-employment-information-chapter/employersPages.js
Comment (tjheffner) [reply src/applications/accreditation/21a/pages/03-employment-information-chapter/employersPages.js:196]: I had originally had them as their own object (and used `object.keys()` for the schema), but the checkbox group wasn't rendering properly, will give it another shot.

File: src/applications/accreditation/21a/pages/04-education-history-chapter/educationalInstitutionsPages.js
Comment (tjheffner) [reply src/applications/accreditation/21a/pages/04-education-history-chapter/educationalInstitutionsPages.js:25]: LGTM, makes sense that any fields conditional required should also be part of a complete item when present :)

**Review by williamphelps13 — CHANGES_REQUESTED** [review]
I'll review this in detail this morning! Can you add the keys of the new fields to this file with mock data: src/applications/accreditation/21a/tests/fixtures/data/[test-data.json](https://github.com/department-of-veterans-affairs/vets-website/blob/a7f1721ae72c398183d8625837f3745f6630fe2e/src/applications/accreditation/21a/tests/fixtures/data/test-data.json). Then run `yarn cy:open` and ensure our 21a e2e submits?

**Review by williamphelps13 — CHANGES_REQUESTED** [review]
One more minor tweak then LGTM

File: src/applications/accreditation/21a/pages/04-education-history-chapter/educationalInstitutionsPages.js
Comment (tjheffner) [reply src/applications/accreditation/21a/pages/04-education-history-chapter/educationalInstitutionsPages.js:82]: ugh yeah, it doesn't seem like required is being passed through the UI pattern properly. I do see the underlying `VaSelectField` has a required propType, so it shouldn't be a huge lift to wire that together correctly

**Review by williamphelps13 — APPROVED** [review]
Awesome job, Tanner!

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**
